PD-552: cli article style and template update (#2034)
truenas/documentation: content/SCALE/SCALECLIReference/Account/CLIuser.md

The `username`, `full_name`, `primary_group`, and `password_status` are all required properties for user creation.
For more details, see the table below in **Create Command Properties**.

{{< expand "Create Command Properties" "v" >}}

{{< include file="AccountUserProperties.md" type="page" >}}

{{< /expand >}}

Enter the command string with property arguments using the `=` delimiter to separate property and value, then press <kbd>Enter</kbd>. 
For example, `uid=3000`.
The command returns a blank line.

To confirm the user is created, use [`get_user_obj`](#get_user_obj-command) or go to [**Credentials > Local Users**]({{< relref "managelocalusersscale.md" >}}) in the SCALE Web UI.

#### Usage
{{< hint type=note >}}
This command contains the minimum required properties to successfully create a user.
{{< /hint >}}

From the CLI prompt, enter:

<code>account user create username=<i>testuser</i> full_name="<i>Test User</i>" group_create=<i>true</i> password=<i>passwort1234</i></code>

Where:
* *testuser* is the desired username

> * *Test User* is the user full name for the user

Reviewer: I discussed options for describing the full name property with Tim recently, looking for something a little less term-in-the-definition. We settled on something like:

```suggestion
* *Test User* is a descriptive name for the user
```

Reviewer: This is a bigger issue than just this one article, but for that field/property in general we need to come up with a more useful description for this one. "**Full name** is a full name for the user" and other versions of that don't actually tell me anything as a reader because I'm just getting the term repeated in the definition.

> * *true* is either true or false. True creates the group at the same time, false does not create the group.

Reviewer: For consistency, you are not punctuating the others. Otherwise add punctuation to the others (and in group create as well)

```suggestion
* *true* is either true or false. True creates the group at the same time, false does not create the group
```

Author: why is the period missing?

Reviewer: see comment above suggested change

* *passwort1234* is the password for the user

{{< expand "Command Example" "v" >}}
```
account user create username=testuser full_name="Test User" group_create=true password=passwort1234

```
{{< /expand >}}
{{< /expand >}}

### Delete Command

The `delete` command removes an existing user account from the system.

{{< expand "Using the Delete Command" "v" >}}
#### Description
The `delete` command has one required property, `id`.
Enter the command string, with the property argument using the `=` delimiter to seprate the property and value, then press <kbd>Enter</kbd>.
The command returns a blank line.

To confirm the user is deleted, use [`get_user_obj`](#get_user_obj-command) or go to [**Credentials > Local Users**]({{< relref "managelocalusersscale.md" >}}) in the SCALE Web UI.

By default, the `user delete` command also deletes the primary group for the user if it is not used by another user.

> Use the `delete_group` property set to `false` to retain the primary group for the user.

Reviewer: ```suggestion
Use the `delete_group` property set to `false` to retain the primary group.
```

#### Usage

From the CLI prompt, enter:

<code>account user delete id=<i>3001</i></code>

Where *3001* is the user ID number for the account.

{{< expand "Command Example" "v" >}}
```
account user delete id=3001
```
{{< /expand >}}

Use the `delete_group` option to keep the primary group of the user.

{{< expand "Command Example" "v" >}}
```
account user delete id=3001 options={"delete_group":false}
```
{{< /expand >}}
{{< /expand >}}

### Get_Instance Command

The `get_instance` command retrieves information about a user such as their username, UID (User ID), group membership, permissions, and other relevant attributes.

{{< include file="/_includes/CLI/CLICommandWIP.md" type="page" >}}

{{< expand "Using Get_Instance Command" "v" >}}
#### Description
The `get_instance` command has one requireproperties, `id`.
Enter property arguments with the `=` delimiter separating property and value.
Enter the command string, then then press <kbd>Enter</kbd>.
The command returns a table of information about the UID entered.

#### Usage
From the CLI prompt, enter:

<code>account user get_instance id=<i>1</i></code>

> Where *1* is the UID for the user, in this case root.

Reviewer: same issue as in group. 1 is the database id for root, 0 is the uid.

### Get_User_Obj Command

The `get_user_obj` command returns a table containing information from **struct passwd** for the user and bypasses user cache.

{{< expand "Using the Get_User_Obj Command" "v" >}}
#### Description
The `get_user_obj` command has one required property, `get_user_obj` that expects you to enter property arguments formatted as an array.
Enter the command string, using `=` followed by the curly brackets that enclose the property argument. 
You can enter either the `uid` or `username` property in this command string.
Enter the property enclosed in double quotes, then the `:` delimiter followed by the value enclosed in double quotes, with no space after the colon. Press<kbd>Enter</kbd>.
The command returns a table for the UID that includes the **pw_name**, **pw_uid**, **pw_gid**, **pw_gecos**, **pw_dir**, and **pw_shell** values.

#### Usage

From the CLI prompt, enter:

> <code>account user get_user_obj get_user_obj={"uid": <i>3001</i>}</code>

Reviewer: ```suggestion
<code>account user get_user_obj get_user_obj={"uid":<i>3001</i>}</code>
```

Where *3001* is the id number for the targeted account.

Alternately enter:

<code>get_user_obj={"username":"<i>testuser</i>"}</code>

Where *testuser* is the username of the targeted account.

{{< expand "Command Example" "v" >}}
<code>

> account user get_user_obj get_user_obj={"username": "<i>testuser</i>"}

Reviewer: ```suggestion
account user get_user_obj get_user_obj={"username":"<i>testuser</i>"}
```

+-----------+--------------+
|   pw_name | testuser     |
|    pw_uid | 3000         |
|    pw_gid | 65534        |
|  pw_gecos | Test User    |
|    pw_dir | /nonexistent |
|  pw_shell | /usr/bin/zsh |
+-----------+--------------+
</code></pre>
{{< /expand >}}
{{< /expand >}}

### Has_Local_Administrator_Set_Up Command

The `has_local_administrator_set_up` command returns whether a local administrator account with a valid password exists.

{{< expand "Using the Has_Local_Administrator_Set_Up Command" "v" >}}
#### Description
The `has_local_administrator_set_up` command does not require entering properties or values.
Enter the command, then press<kbd>Enter</kbd>.

> The command returns true if the admin account has a password, false if it does not.

Reviewer: Technically I think it returns true if either admin or root exist. I don't think the password matters.

#### Description
The `setup_local_administrator` command has two required properties, `username`, `password`. 
The `options` property is not required and is entered as an array.
Enter property arguments using the `=` delimiter to separate the property and value, then enter <kbd>Enter</kbd>.
The command returns a blank line.
<!-- Commenting this out until it is verified in cobia.
Enter the CLI from the [Console setup menu]({{< relref "ConsoleSetupMenuScale.md" >}}) before configuring a local administrator or root user account. -->
{{< hint type=note >}}
`setup_local_administrator` accepts only "admin" or "root" for **username**.
It is recommended to use "admin." Root user access is a deprecated method.
{{< /hint >}}

To confirm the administrator is set up, use [`has_local_administrator_set_up`](#has_local_administrator_set_up-command), [`get_user_obj`](#get_user_obj-command), or log in to the SCALE Web UI using the admin username and password.

#### Usage

From the CLI prompt, enter:

<code>account user setup_local_administrator username:<i>admin</i> password:<i>passw0rt</i></code>

Where:
* *passw0rt* is the password for the admin account.

> * *admin* is the administration user. If the `username` value is *root* it sets up the root user.

Reviewer: I think the deleted hint, or a version of it, should still be here. My intention there was to explain that the command will fail unless username is either "admin" or "root". You can't create an administrator named anything else using this command

Author: details like a hint about entering a command belong in the Description section not in the Usage where we just demonstrate entering the command. I covered this in the description. the reason admin is a variable is because you can enter one or the other which makes it a variable. If neither the admin or root user are created then you can enter either one. Again, details about this belong in the Description text not in the Usage.,

### Shell_Choices Command

The `shell_choices` command returns the shell choices available to user accounts.
The `shell_choices` command returns the shell choices available to user accounts. 

{{< expand "Using the Shell_Choices Command" "v" >}}

#### Description

Enter the command with no additional properties to return shell choices available to all users.

> Additional shell choices are available to users with administrative privileges (members of the **builtin_administrators** group).

Reviewer: I think we do need to offer both versions of the command in some form. If the user has administrative privledges they can access the additional options, but if not they cannot. It isn't intuitive to find those additional options or to know they are there unless we explain it. 

```
[qe-scale-04]> account user shell_choices group_ids=["builtin_administrators"]
+----------------------+-----------------+
|    /usr/sbin/nologin | nologin         |
|         /usr/bin/cli | TrueNAS CLI     |
| /usr/bin/cli_console | TrueNAS Console |
|        /usr/bin/bash | bash            |
|       /usr/bin/rbash | rbash           |
|        /usr/bin/dash | dash            |
|          /usr/bin/sh | sh              |
|        /usr/bin/tmux | tmux            |
|         /usr/bin/zsh | zsh             |
+----------------------+-----------------+
```
```
[qe-scale-04]> account user shell_choices
+-------------------+---------+
| /usr/sbin/nologin | nologin |
|     /usr/bin/bash | bash    |
|    /usr/bin/rbash | rbash   |
|     /usr/bin/dash | dash    |
|       /usr/bin/sh | sh      |
|     /usr/bin/tmux | tmux    |
|      /usr/bin/zsh | zsh     |
+-------------------+---------+
```

Author: Short answer is no. We can revisit how we want to do specific examples, like a command string that configures HA networking or creating a VLAN or LAGG interface and then include the command examples for this, but it is not necessary in this case. One example command output is sufficient.

Reviewer: Maybe not a command example, but at least a sentence in the description

+-------------------+---------+
```
{{< /expand >}}
{{< /expand >}}

### Update Command
The `update` command updates the attributes of an existing user. 

{{< expand "Using the Update Command" "v" >}}
#### Descripton

> The `update` command uses the the same properties as the [`create`](#create-command) command.

Reviewer: ```suggestion
The `update` command uses the same properties as the [`create`](#create-command) command.
```

{{< expand "Update Command Properties" "v" >}}

{{< include file="AccountUserProperties.md" type="page" >}}

{{< /expand >}}

The required property is `uid_or_username`.
Enter property arguments with the `=` delimiter separating property and values, then press <kbd>Enter</kbd>.
The command returns a blank line.

To confirm the user is updated, use [`get_user_obj`](#get_user_obj-command) or go to [**Credentials > Local Users**]({{< relref "managelocalusersscale.md" >}}) in the SCALE Web UI.

#### Usage
From the CLI prompt, enter:

<code>account user update uid_or_username=<i>3001</i></code>

Where *3001* is the UID or username of the account.

To update the email address for an account, enter:

<code>account user update uid_or_username=<i>3001</i> email="<i>mail&#64;email.com</i>"</code>

Where:
* *3001* is the UID or username of the account to update.
* *mail&#64;email.com* is the update made to the email for the user specified.

{{< expand "Command Example" "v" >}}
```
account user update uid_or_username=3001

```
{{< /expand >}}
{{< /expand >}}

### Verify_Twofactor_Token Command

> The `verify_twofactor_token` command verifes the token for the username specified is successfully authenticated.

Reviewer: ```suggestion
The `verify_twofactor_token` command verifies the token for the username specified is successfully authenticated.
```
